src/main/java/marquez/MarquezApp.java

```diff
@@ -102,13 +102,20 @@ private void migrateDbOrError(@NonNull MarquezConfig config) {
     // requiring we handle the throwable and apply a repair on the database to fix any
     // issues before terminating.
     try {
+      log.info("Migrating database...");
       flyway.migrate();
-    } catch (FlywayException e) {
-      log.error("Failed to apply migration to database.", e.getMessage());
-      log.info("Repairing failed database migration...", e.getMessage());
-      flyway.repair();
-
-      log.info("Successfully repaired database, stopping app...");
+      log.info("Successfully migrated database.");
+    } catch (FlywayException errorOnDbMigrate) {
+      log.error("Failed to apply migration to database.", errorOnDbMigrate);
+      try {
+        log.info("Repairing failed database migration...");
+        flyway.repair();
+        log.info("Successfully repaired database.");
+      } catch (FlywayException errorOnDbRepair) {
+        log.error("Failed to apply repair to database.", errorOnDbRepair);
+      }
+
+      log.info("Stopping app...");
       // The throwable is not propagating up the stack.
       onFatalError(); // Signal app termination.
```

Comment:
I think a few unit tests would be good to verify error behavior and what is called when specific exceptions are thrown.